pulser/sequence.py

```diff
@@ -833,6 +781,33 @@ def target(
 
         self._target(qubits, channel)
 
+    @_verify_parametrization
+    def target_index(
+        self,
+        qubits: Union[int, Iterable[int], Parametrized],
+        channel: Union[str, Parametrized],
+    ) -> None:
+        """Changes the target qubit of a 'Local' channel.
+
+        Args:
+            qubits (Union[int, str, Iterable]): The new target for this
+                channel. Must correspond to a qubit index (a number between 0
+                 and the number of qubits, which is determined at build time
+                 when using a MappableRegister) or an iterable
+                of qubit indices, when multi-qubit addressing is possible.
+            channel (str): The channel's name provided when declared. Must be
+                a channel with 'Local' addressing.
+        """
+        if not self.is_parametrized():
+            raise RuntimeError(
+                f"Sequence.{self.target_index.__name__} can't be called in"
+                " non parametrized sequences."
+            )
+
```

Comment:
Do we need to restrict this to when the sequence is parametrized? I agree that it makes more sense to use it when there are variables involved, but it doesn't have to be the case.

Reply:
I think we agreed to allow the index only for parametrized sequences, because non-parametrized sequences can still use a mappable register, so it will fail to compute the indices.

Reply:
I think we agreed on allowing the index only for parametrized sequences, but I can't remember why.

Reply:
Or rather, `self.is_register_mappable() and not self.is_parametrized()`

Reply:
Do you mean, only if `self.is_register_mappable() and not self.is_parametrized()`? Since an index target makes little sense without parametrization, wouldn't it be simpler to keep the only parametrization condition? We may also arbitrarily mark the sequence as parametrized in such cases, even if there are no parameters, but this doesn't waive out the complexity.

Reply:
I think it is about the same complexity for the support, but I had in mind the complexity of the documentation for the users. Maybe the best would be to automatically switch the sequence to parametrized every time `target_index` is called with `self.is_register_mappable()`. Thus, it will always work, but will simply switch to parametrized in a small documented edge case.